api/src/services/user-groups/create-service.ts

```diff
@@ -25,14 +26,34 @@ export class CreateService extends BaseService {
       throw new Error("Group ID is required")
     }
 
-    const userGroup = await UserGroup.create({
-      creatorId: this.currentUser.id,
-      userId,
-      groupId,
-      ...optionalAttributes,
+    return db.transaction(async () => {
+      // _TODO_ can I eager load the user and group?
+      const userGroup = await UserGroup.create({
+        creatorId: this.currentUser.id,
+        userId,
+        groupId,
+        ...optionalAttributes,
+      })
```

Comment:
I suggest removing this comment and doing
```ts
const userGroup = await UserGroup.create({
    ...optionalAttributes,
  creatorId: this.currentUser.id,
  userId,
  groupId,
})
await userGroup.reload({ include: [....] })
const { user, group } = userGroup
```

Note that optional attributes should go first, otherwise you potentially allow the user to override the "creatorId" field.